src/ram/transform/MakeIndex.cpp

```diff
@@ -274,6 +122,60 @@ Own<RamCondition> MakeIndexTransformer::constructPattern(const std::vector<std::
         }
     };
 
+    // transform condition list so that every strict inequality becomes a weak inequality + filter
+    // e.g. Tuple[level, element] < <expr> --> Tuple[level, element] <= <expr> and Tuple[level, element] !=
+    // <expr>
+    std::vector<std::unique_ptr<RamCondition>> toAppend;
+    auto it = conditionList.begin();
+    while (it != conditionList.end()) {
+        auto& cond = *it;
+        if (auto* binRelOp = dynamic_cast<RamConstraint*>(cond.get())) {
+            bool indexable = false;
+            size_t element = 0;
+
+            if (isStrictIneqConstraint(binRelOp->getOperator())) {
+                if (const auto* lhs = dynamic_cast<const RamTupleElement*>(&binRelOp->getLHS())) {
+                    const RamExpression* rhs = &binRelOp->getRHS();
+                    if (lhs->getTupleId() == identifier && rla->getLevel(rhs) < identifier) {
+                        indexable = true;
+                        element = lhs->getElement();
+                    }
+                }
```

Comment:
I don't really know how the transformer works. But I feel like you should include an extra test here:
1.  check if we are in the interpreter mode
2.  check if the target operation is a signed/float one. 
If both true, you should NOT mark `indexable` to true, as it will confuse the aggregates.